### DISCUSSION TITLE
Continue to create after halting

### DISCUSSION THREAD

**Discussion by PieterxJan** [filamentphp/filament#5430] [General]
Hi,

I'd like to implement some kind of "Are you sure?" notification/modal when creating something. I'm able to halt the creation, close the notification, but I have no idea how to continue the action (if that is even possible?)

```php
protected function beforeCreate(): void
{
    Notification::make()
        ->warning()
        ->title('Warning')
        ->body('Are you sure you want to continue to create this?')
        ->persistent()
        ->actions([
            Action::make('continue')->label('Yes, create')->button(),
            Action::make('cancel')->close()
        ])
        ->send();

    $this->halt();
}
```

Thanks in advance!

**Comment by citroid** [comment]
Hi, 

I would be interested in this topic as well as I have a similar use case.
My application performs a check for possible duplicates before a new resource ist created.
If candidates for such duplicates have been found URLs for these records should be displayed in the notification so the user can review them. The user should then decide if he wants to proceed or cancel, which leads me to the same problem: how can I continue with the resource creation once it has been halted?

Currently my only idea would be to introduce a hidden form input which could be used to indicate an override of the duplicate check. By triggering another form submit the resource should then be created as it normally would. 

Is there a more intuitive and elegant approach?

**Comment by dev** [comment]
Any updates on this? I'm currently working on a project where this feature would be useful as well.

**Comment by akulmambetov** [comment]
Any updates? @danharrin

**Comment by sahapranta** [comment]
I am managing using this approach, but it would be better if there is option to procced or cancel after halt
```php
->before(function (CreateAction $action, array $data) {
    $key = "agent_create_" . $data['customer_id'];
    if (Cache::has($key)) {
        Cache::forget($key);
    } else if (Model::where('customer_id', $data['customer_id'])->exists()) {
        Notification::make()
            ->warning()
            ->title('Customer already exists!')
            ->body('This customer already exists. Do you want to proceed?')
            ->duration(2000)
            ->send();
        Cache::remember($key, now()->addMinute(), fn() => true);
        $action->requiresConfirmation();
        $action->halt();
    }
})
```
